media/starboard/starboard_cdm.cc

```diff
@@ -1,254 +1,196 @@
-    SessionUpdateKeyStatusesCallback session_update_key_statuses_callback,
-    SessionClosedCallback session_closed_callback) {
-  DCHECK(task_runner_->RunsTasksInCurrentSequence());
-  return std::unique_ptr<DrmSystem::Session>(new Session(
-      this, session_update_key_statuses_callback, session_closed_callback));
+  NOTREACHED() << "Unexpected SbDrmSessionRequestType " << type;
 }
 
-bool DrmSystem::IsServerCertificateUpdatable() {
-  DCHECK(task_runner_->RunsTasksInCurrentSequence());
-  DCHECK(is_valid());
-
-  if (SbDrmIsServerCertificateUpdatable(wrapped_drm_system_)) {
-    LOG(INFO) << "SbDrmSystem (" << wrapped_drm_system_
-              << ") supports server certificate update";
-    return true;
+CdmKeyInformation::KeyStatus ToCdmKeyStatus(SbDrmKeyStatus status) {
+  switch (status) {
+    case kSbDrmKeyStatusUsable:
+      return CdmKeyInformation::KeyStatus::USABLE;
+    case kSbDrmKeyStatusExpired:
+      return CdmKeyInformation::KeyStatus::EXPIRED;
+    case kSbDrmKeyStatusReleased:
+      return CdmKeyInformation::KeyStatus::RELEASED;
+    case kSbDrmKeyStatusRestricted:
+      return CdmKeyInformation::KeyStatus::OUTPUT_RESTRICTED;
+    case kSbDrmKeyStatusDownscaled:
+      return CdmKeyInformation::KeyStatus::OUTPUT_DOWNSCALED;
+    case kSbDrmKeyStatusPending:
+      return CdmKeyInformation::KeyStatus::KEY_STATUS_PENDING;
+    case kSbDrmKeyStatusError:
+      return CdmKeyInformation::KeyStatus::INTERNAL_ERROR;
   }
-  LOG(INFO) << "SbDrmSystem (" << wrapped_drm_system_
-            << ") doesn't support server certificate update";
-  return false;
 }
 
-void DrmSystem::UpdateServerCertificate(
-    const uint8_t* certificate, int certificate_size,
-    ServerCertificateUpdatedCallback server_certificate_updated_callback) {
-  DCHECK(task_runner_->RunsTasksInCurrentSequence());
-  DCHECK(IsServerCertificateUpdatable());
-  DCHECK(is_valid());
-
-  if (!certificate) {
-    LOG(ERROR) << "Updating server with NULL certificate";
-    return;
-  }
+}  // namespace
+
+StarboardCdm::StarboardCdm(
+    const CdmConfig& cdm_config,
+    const SessionMessageCB& message_cb,
+    const SessionClosedCB& closed_cb,
+    const SessionKeysChangeCB& keys_change_cb,
+    const SessionExpirationUpdateCB& expiration_update_cb)
+    : message_cb_{message_cb},
+      closed_cb_{closed_cb},
+      keys_change_cb_{keys_change_cb},
+      expiration_update_cb_{expiration_update_cb},
+      sb_drm_(SbDrmCreateSystem(cdm_config.key_system.c_str(),
+                                this,
+                                OnSessionUpdateRequestGeneratedFunc,
+                                OnSessionUpdatedFunc,
+                                OnSessionKeyStatusesChangedFunc,
+                                OnServerCertificateUpdatedFunc,
+                                OnSessionClosedFunc)),
+      task_runner_(base::SequencedTaskRunner::GetCurrentDefault()) {}
+
+StarboardCdm::~StarboardCdm() {
+  DVLOG(1) << __func__;
+}
```

Comment:
Just want to double check where is `SbDrmDestroySystem()` called?

Reply:
ah, missed that, thanks.